test/multiverse/suites/logstasher/logstasher_instrumentation_test.rb

```diff
@@ -0,0 +1,123 @@
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+require 'ostruct'
+
+class LogStasherInstrumentationTest < Minitest::Test
+  include MultiverseHelpers
+
+  def setup
+    @written = StringIO.new
+
+    # Give LogStasher's setup method a new place to write logs to, as well as
+    # skip a config check, controller_monkey_patch, that otherwise causes an error
+    LogStasher.setup(OpenStruct.new(logger_path: @written, controller_monkey_patch: false))
+
+    # required for build_logstasher_event method
+    LogStasher.field_renaming = {}
+    @aggregator = NewRelic::Agent.agent.log_event_aggregator
+
+    NewRelic::Agent.instance.stats_engine.reset!
+    NewRelic::Agent.instance.log_event_aggregator.reset!
+  end
+
+  def teardown
+    NewRelic::Agent.instance.stats_engine.reset!
+    NewRelic::Agent.instance.log_event_aggregator.reset!
+  end
+
+  def json_log_hash
+    {
+      :identifier => 'dinosaurs/_dinosaur.html.erb',
+      :name => 'render_partial.action_view',
+      :request_id => '01234-abcde-56789-fghij',
+      'source' => '127.0.0.1',
+      'tags' => [],
+      '@timestamp' => '2024-06-24T23:55:59.497Z'
+    }
+  end
+
+  def test_level_is_recorded
+    in_transaction do
+      LogStasher.build_logstash_event({'level' => :info, 'message' => 'hi there'}, ['log'])
+    end
+    _, events = @aggregator.harvest!
+
+    assert_equal :info, events[0][1]['level']
+    assert_metrics_recorded(%w[Logging/lines/info])
```

Comment:
The capitalization of "info" is different from the capitalization of "UNKNOWN" in the following test. I believe this means that we'll get different metrics and counts for "Logging/lines/info" and "Logging/lines/INFO".

I don't see this specifically mentioned in the spec, but the reference always has a capitalized severity. I think we want all the severity levels to be capitalized within the metric name regardless of how they're spelled when we perform the count in order to have an accurate count of that log severity category in general. 

I think we got away with not caring about this for the Ruby logger since it has consistent severity capitalization in its constants, but this issue may exist for that instrumentation too.

Reply:
Thank you! This is a good call out. I've gone ahead and made sure severity levels are always capitalized https://github.com/newrelic/newrelic-ruby-agent/pull/2559/commits/45e6bc4deb266e90fcc683cefe42a1c53cc655af